torch/_inductor/compile_fx_async.py

```diff
@@ -211,9 +211,8 @@ def __init__(
     ) -> None:
         self._fast_output_code = fast_output_code
         self._optimized_output_code = None
-        self._progression_futures = list(progression_futures)
+        self._progression_futures = deque(progression_futures)
```

Comment:
nit: Above (github won't let me comment on unchanged lines):
```
    progression_futures: Sequence[Future[_WireProtocolPickledOutput]],
```

Reply:
Problem is then mypy complains about no popleft

```
  Error (MYPY) [attr-defined]
    "Sequence[Future[_WireProtocolPickledOutput]]" has no attribute "popleft"

        268  |
        269  |        # Clear earlier progression futures to free memory
        270  |        for _ in range(stage_index + 1):
    >>> 271  |            self._progression_futures.popleft()
        272  |
        273  |    @override
        274  |    def post_compile(
```

Reply:
I think you misunderstood. I meant the __init__ signature should take Sequence. The instance member should still be deque.